### DISCUSSION TITLE
Issue with fluxcd

### DISCUSSION THREAD

**Discussion by hari12091989** [fluxcd/flux2#2837] [Q&A] [answered]
[deployment.txt](https://github.com/fluxcd/flux2/files/8899796/deployment.txt)
Hi All,

I have deployed fluxcd in Azure Aks.

All the installations are fine. I created some sample yaml files for namespace creation and nginx deployment creations.

After creating yaml files, i have pushed to azure devops repository.

After that reconcilation takes place and revison number changes but my new namespace or deployments are not created.

Fewtimes it worked for sometime(maybe for 15 to 20 mins) again stops working after that.

Can you please help solving my issue.

azureuser@fluxcd-poc-vm:~/Azure-Fluxcd-poc/clusters/fluxcd-poc/flux-system$ flux get  kustomizations --watch
NAME       REVISION      SUSPENDEDREADYMESSAGE                          
flux-systemmaster/e13bfc5False    True Applied revision: master/e13bfc5
[deployment.txt](https://github.com/fluxcd/flux2/files/8899784/deployment.txt)

azureuser@fluxcd-poc-vm:~/Azure-Fluxcd-poc/clusters/fluxcd-poc/flux-system$ kubectl get pods -n default
No resources found in default namespace.

**Comment by somtochiama** [comment]
Is your Flux Kustomization pointing to the directory containing your YAML files and do you have a `spec.targetNamespace` set?
Run `kubectl get deployments` with that namespace instead.

**Reply by hari12091989** [reply]
Hi Somtochi, 
Regarding your answer above, spec.targetNamespace is for Helm right? i am not using helm.
All the other steps mentioned i have tried and it is still not coming up

**Reply by kingdonb** [reply]
There is also a `spec.targetNamespace` on Flux Kustomization:

```
$ k explain kustomization.spec.targetNamespace
KIND:     Kustomization
VERSION:  kustomize.toolkit.fluxcd.io/v1beta2

FIELD:    targetNamespace <string>

DESCRIPTION:
     TargetNamespace sets or overrides the namespace in the kustomization.yaml
     file.
```

**Comment by kingdonb** [comment, marked as answer]
Could you elaborate on exactly what worked and what stops working? Did the controller apply the resource once and then stop? (Are other resources from other Flux Kustomizations being applied? Is this a multi-tenant cluster?)

I would suggest running `flux tree` to be sure that the resources are known to Flux, and `flux build` to ensure that Flux thinks it should apply them. There is more information about these commands in the docs if you have trouble using them.

if you can provide the manifests that reproduce this issue and share them with us then we can help to diagnose. I'm guessing there is something in your manifests which exclude or ignore the files that you have just added. Either in `.sourceignore` or `spec.ignore` of your GitRepository, or excluded because it's not in the `spec.path` of your Flux Kustomization, or excluded by a `kustomization.yaml` which refers to other files but not these.

Flux uses a "recursive autodetection" behavior in the event that you have not written your own `kustomization.yaml` Kustomize overlays. If you have written one, then all resources must be listed in it. There are a half dozen things which can be blocking the apply that are normal and expected, we'd have to check for all of them in order to rule out the possibility that there is no bug.

**Reply by hari12091989** [reply]
Hi Kingdonb,

Please find the details as below:
My cluster name is fluxcd-poc which is created in Azure(AKS). It is not a multi-tenant cluster. It is only a single cluster.

I am logged into the Aks cluster through a Windows VM, through bastion.

The yaml file i deployed is a very simple nginx deployment, which is attached here.

After changing the details (for example) : I increased the replicas from 2 to 3. I commited the new yaml file and pushed it to my azure repo.

After which i ran the command **flux get kustomizations --watch**

It shows output as:

**azureuser@Windows-VM MINGW64 ~/Azure-Fluxcd-Poc/clusters/fluxcd-poc/flux-system (master)
$ flux get kustomizations --watch
NAME            REVISION        SUSPENDED       READY   MESSAGE
flux-system     master/9ca21c9  False           True    Applied revision: master/9ca21c9****        

Once i push my file to my repo, my commit id in this case 9ca21c9** matches with my repo commit id
Also attached the commit id in azure repo

but when i look for a created pod/deployment in default namespace, nothing is created as shown below
 
**$ kubectl get pods -n default
No resources found in default namespace.**

Also i am not changing on the kustomization yaml files.

The below is the commits and push details from my azure-vm to the azure-repo.

**azureuser@Windows-VM MINGW64 ~/Azure-Fluxcd-Poc/clusters/fluxcd-poc/flux-system (master)
$ git commit -m "Increasing the replicas"
[master 9ca21c9] Increasing the replicas
 1 file changed, 1 insertion(+), 1 deletion(-)

azureuser@Windows-VM MINGW64 ~/Azure-Fluxcd-Poc/clusters/fluxcd-poc/flux-system (master)
$ git push origin master
Enumerating objects: 11, done.
Counting objects: 100% (11/11), done.
Delta compression using up to 2 threads
Compressing objects: 100% (3/3), done.
Writing objects: 100% (6/6), 427 bytes | 85.00 KiB/s, done.
Total 6 (delta 2), reused 0 (delta 0), pack-reused 0
remote: Analyzing objects... (6/6) (135 ms)
remote: Storing packfile... done (38 ms)
remote: Storing index... done (52 ms)
To ssh.dev.azure.com:v3/HariharanR0582/Azure-Fluxcd-Poc/Azure-Fluxcd-Poc
   93e8f01..9ca21c9  master -> master

The flux pods are all running fine and without any erros
$ kubectl get pods -n flux-system
NAME                                       READY   STATUS    RESTARTS   AGE
helm-controller-5985c795f8-strpb           1/1     Running   0          7d22h
kustomize-controller-78cb866b68-p8lrb      1/1     Running   0          7d22h
notification-controller-6b7d7485fc-7gsxp   1/1     Running   0          7d22h
source-controller-644c69fbf7-jqq9v         1/1     Running   0          7d22h

The logs which i got after the yaml file has been pushed to azure Repo:

kubectl logs -f source-controller-644c69fbf7-jqq9v -n flux-system
**{"level":"info","ts":"2022-06-29T13:15:47.476Z","logger":"controller.gitrepository","msg":"no changes since last reconcilation: observed revision 'master/9ca21c9ed0270752472e00a2845f3b4fb10e1849'","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"GitRepository","name":"flux-system","namespace":"flux-system"}
{"level":"info","ts":"2022-06-29T13:16:17.799Z","logger":"controller.gitrepository","msg":"no changes since last reconcilation: observed revision 'master/9ca21c9ed0270752472e00a2845f3b4fb10e1849'","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"GitRepository","name":"flux-system","namespace":"flux-system"}
{"level":"info","ts":"2022-06-29T13:16:48.006Z","logger":"controller.gitrepository","msg":"no changes since last reconcilation: observed revision 'master/9ca21c9ed0270752472e00a2845f3b4fb10e1849'","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"GitRepository","name":"flux-system","namespace":"flux-system"}**

Please let me know, if anything else is required from my side and i can share those details with you

**Reply by kingdonb** [reply]
I think I see the problem.

You have committed your deployment in the directory: `clusters/fluxcd-poc/flux-system`

In this directory, there is a `kustomization.yaml` manifest for the Kustomize overlay which says "load only these files" – it usually lists `gotk-sync.yaml` and `gotk-components.yaml` – if you add more YAML files to this directory, they will not be loaded by Flux.

You are meant to create more directories in `clusters/fluxcd-poc/` and add your YAML files there. Since the Flux Kustomization path is set to this directory, it will slurp up all the YAMLs in that directory according to the behavior described in the FAQ, namely this:

`kustomize build --autodetect --recursive`

Try it in your repo, it will generate a `kustomization.yaml` so you can see what Flux does under the hood (but I don't recommend to commit the resulting `kustomization.yaml` unless you want to break autodetection for the rest of the cluster path from here on out!)

The solution I would recommend is just to move your `deployment.yaml` outside of the `flux-system` directory. Or better yet, add more indirection. You can create additional Flux kustomizations and sync them from other directories outside of the path `clusters/fluxcd-poc` – there are some suggestions about how to structure your repository here in the Flux docs, a guide: https://fluxcd.io/docs/guides/repository-structure/

**Reply by hari12091989** [reply]
HI Kingdon barrett,

I tried creating a new directory and placed my deployment.yaml files inside the new directory, outside of the flux-system directory and fluxcd worked flawlessly without any issues.

Thank you so much for your help in this issue.

**Comment by hari12091989** [comment]
[deployment.txt](https://github.com/fluxcd/flux2/files/9010779/deployment.txt)
<img width="372" alt="Commit capture" src="https://user-images.githubusercontent.com/57386786/176447748-33cbaee6-134e-4080-b9f2-cfe4e9cd06da.PNG">
